lib/exporter/OrgFileExporter.js

```diff
@@ -0,0 +1,214 @@
+        });
+      },
+      (next) => {
+        const dir = `${path.dirname(filepath)}/services`;
+        this._exportServices({ dir, orgId: rawData.id }, (err, exportedServices) => {
+          if (err) {
+            return next(err);
+          }
+
+
+          exportData.services = exportedServices;
+          next();
+        });
+      }
+    ], (err) => {
+      if (err) {
+        return done(err);
+      }
+
+      this.cliManager.log(LogLevel.DEBUG, `Writing configuration to file: ${filepath}`);
+      try {
+        fs.writeFileSync(filepath, JSON.stringify(exportData, null, 4));
+      } catch (ex) {
+        return done(ex);
+      }
+
+      done(null, exportData);
+    });
+  }
+
+  /**
+   * Exports the org's apps if any.
+   * @param {Object} options
+   * @param {String} options.dir Directory where to export the apps.
+   * @param {String} options.orgId ID of the org.
+   * @param done
+   * @private
+   */
+  _exportApps(options, done) {
+    const result = {};
+    let apps;
+
+    async.series([
+      (next) => {
+        this.applicationsService.getByOrg(options.orgId, (err, data) => {
+          if (err) {
+            return next(err);
+          }
+
+          apps = data;
+          next();
+        });
+      },
+      (next) => {
+        if (isEmpty(apps)) {
+          return setImmediate(next);
+        }
+
+        mkdirp(`${options.dir}/a`, next); // mkdirp expects filePath to create the directory
+      },
+      (next) => {
+        async.eachSeries(
+          apps,
+          (currApp, cb) => {
+            const appName = currApp.name;
+            const fileName = `${appName}.json`;
+            const filePath = `${options.dir}/${appName}/${fileName}`;
+            this.appExporter.exportApp({ file: filePath, [AppOptionsName.APP]: currApp.id }, (err) => {
+              if (err) {
+                return cb(err);
+              }
+
+              result[appName] = `${ConfigFiles.FILE_REFERENCE_PREFIX}./${path.basename(options.dir)}/${appName}/${fileName}`;
+              cb();
+            });
+          },
+          next
+        );
+      }
+    ], (err) => {
+      done(err, result);
+    });
+  }
+
+  /**
+   * Exports the org's services if any.
+   * @param {Object} options
+   * @param {String} options.dir Directory where to export the services.
+   * @param {String} options.orgId ID of the org.
+   * @param done
+   * @private
+   */
+  _exportServices(options, done) {
+    const result = {};
+    let services;
+
+    async.series([
+      (next) => {
+        this.servicesService.getAllByDomainType(DomainTypes.ORG, { id: options.orgId }, (err, data) => {
+          if (err) {
+            return next(err);
+          }
+
+          services = data;
+          next();
+        });
+      },
+      (next) => {
+        if (isEmpty(services)) {
+          return setImmediate(next);
+        }
+
+        mkdirp(`${options.dir}/a`, next); // mkdirp expects filePath to create the directory
```

Comment:
Shouldn't we put this inside the exportService function? It is best for the function itself to make sure the specified path exists, as opposed to making sure it is created in every place that you invoke the function.

Reply:
Here, we create a service directory (only in case there are services to be exported) and we tell the exportService function where to save the exported service. The same function is used on the 'service export' command, so it expects a file path.

Reply:
So, there is an exportService() function that accepts a filePath as an argument. There is a hidden requirement though - the directory for the given filePath must already be created, otherwise the function will fail. Is that correct?
What I am suggesting is making sure the necessary folders are created inside the exportService() function, so that there is no hidden requirements.

Reply:
What you call a 'hidden requirement', I call 'expected behavior' :grin:
I think we have a slight communication gap. When I implemented the export commands I thought that the user must specify a path where all the directories already exist. I see now that when you implemented the env export, you thought that it is fine if a path with non-existing directories is specified...
Can I fix it here or should I create a ticket?

Reply:
You should create a ticket. Doesn't mean you cannot fix the ticket in this PR. Up to you.